extensions/ql-vscode/src/remote-queries/analyses-results-manager.ts

```diff
@@ -139,6 +146,6 @@ export class AnalysesResultsManager {
   }
 
   private isAnalysisDownloaded(analysis: AnalysisSummary): boolean {
-    return this.analysesResults.some(x => x.nwo === analysis.nwo);
+    return (this.analysesResults.get(analysis.downloadLink.queryId) || []).some(x => x.nwo === analysis.nwo);
```

Comment:
Here and in a few other places we do `this.analysesResults.get(queryId)` .. I think it'd be easier to read if we had a `getQueryResults` and did

```suggestion
  return this.getQueryResults(queryId).some(x => x.nwo === analysis.nwo);
```

where `getQueryResults` is:
```ts
private getQueryResults(queryId: string) { 
  return this.analysesResults.get(analysis.downloadLink.queryId) || []
}
```

Reply:
There is already the `getAnalysesResults`, which is public. It makes a copy of the array before returning it. I can't use this method internally, since there is one place that relies on using the original array. Instead, I will make a `internalGetAnalysesResults` method that returns the results without copying them.